torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -153,6 +153,7 @@ def apply_tp(
             "tok_embeddings": RowwiseParallel(
                 input_layouts=Replicate(),
                 output_layouts=Shard(1),
+                use_local_output=False,
```

Comment:
Curious why is `use_local_output` related to sequence length?

Reply:
When `use_local_output = True`, the output of this layer is a sharded `torch.Tensor`, which has no idea of the global shape. E.g. a sequence of length 3 sharded with TP degree 2 will result in (a length-2 subsequence on rank 0, a length-1 subsequence on rank 1). When we later perform all-gather on the sharded sequence, it will infer the global shape based on the assumption the tensor is evenly sharded, causing shape mismatch and hang.

When `use_local_output = False`, the DTensor object keeps track of the global shape.

DTensor has API to input shapes manually (instead of inferring); it's just the TP API currently don't support it yet.